### ISSUE TITLE
Generate `hotwired/turbo-site` content from `hotwired/turbo` Source Code

### ISSUE THREAD

**Issue by seanpdoyle** [hotwired/turbo#997]
Since the team's [official stance](https://github.com/hotwired/turbo/pull/984#issuecomment-1711491766) is that end-user facing documentation serves as the project's source of truth (for type information and more), it would be extremely valuable to embed the [documentation website](https://github.com/hotwired/turbo-site) into the project's source code.

The [documentation pages for Stimulus](https://github.com/hotwired/stimulus-site) are generated by commits in the project's [docs/](https://github.com/hotwired/stimulus/tree/main/docs) directory. By automatically integrating the two repositories, contributors can propose documentation revisions simultaneously with source code revisions. Without tight integration, it's possible for changes to public interfaces (either through changes to internal JavaScript object properties and functions, Custom Element classes, or changes to `CustomEvent.details`)

What can be done to enable the same integration and lock step for the Turbo project?

**Comment by seanpdoyle** [comment]
In the meantime, I've opened https://github.com/hotwired/turbo-site/pull/145 to document CustomEvent type information that was lost with the removal of TypeScript.

Some CustomEvents are dispatched with references to internally defined Turbo JavaScript objects like [FormSubmission](https://github.com/hotwired/turbo/blob/c44664d6acea224502d440754e2376635a6e9652/src/core/drive/form_submission.js), [FetchRequest](https://github.com/hotwired/turbo/blob/c44664d6acea224502d440754e2376635a6e9652/src/http/fetch_request.js), [FetchResponse](https://github.com/hotwired/turbo/blob/c44664d6acea224502d440754e2376635a6e9652/src/http/fetch_response.js), [FrameElement](https://github.com/hotwired/turbo/blob/c44664d6acea224502d440754e2376635a6e9652/src/elements/frame_element.js), and [StreamElement](https://github.com/hotwired/turbo/blob/c44664d6acea224502d440754e2376635a6e9652/src/elements/stream_element.js).

Without providing type information for those objects, they are prone to change shape in a way that has unintended side-effects related to backward compatibility. If the documentation for those classes resided in the same repository as their source code, it would be easier to coordinate changes in both places at once.

**Comment by brunoprietog** [comment]
I strongly agree with this, is there anything that can be done? Currently there are already several things that have been released and are not documented yet. Having separate repositories I feel exacerbates the problem. @jorgemanrubia @afcapel what do you think?

**Comment by seanpdoyle** [comment]
> What can be done to enable the same integration and lock step for the Turbo project?

@jorgemanrubia @afcapel @kevinmcconnell now that Turbo 8 has been released, I wonder if there's an opportunity to consolidate `@hotwired/turbo-site` into this repository.
